src/NodeTypeResolver/TypeComparator/TypeComparator.php

```diff
@@ -87,10 +84,6 @@ public function arePhpParserAndPhpStanPhpDocTypesEqual(
             $node
         );
 
-        // normalize bool union types
-        $phpParserNodeType = $this->normalizeConstantBooleanType($phpParserNodeType);
-        $phpStanDocType = $this->normalizeConstantBooleanType($phpStanDocType);
```

Comment:
this apparently, make the valid removal skipped for super type:

```
     /** @var string|bool|null */
     private static string|false|null $inputArgSeparator = null;
```

which docblok is super type, I will look more.

Reply:
I added failing test case for valid removal super type https://github.com/rectorphp/rector-src/pull/5588/commits/0c89834ab73ba3384c74d729d227c533cce5b5db

Reply:
Fixed with equal and sub type check https://github.com/rectorphp/rector-src/pull/5588/commits/3690cad412c0a2c9632c128c3c129c573ad79fc0